src/Kiota.Builder/KiotaBuilder.cs

```diff
@@ -851,7 +854,7 @@ private CodeTypeBase CreateModelDeclarations(OpenApiUrlTreeNode currentNode, Ope
                                         var shortestNamespaceName = GetModelsNamespaceNameFromReferenceId(propertyDefinitionSchema.Reference?.Id);
                                         var targetNamespace = string.IsNullOrEmpty(shortestNamespaceName) ? ns : 
                                                                 (rootNamespace.FindNamespaceByName(shortestNamespaceName) ?? rootNamespace.AddNamespace(shortestNamespaceName));
-                                        definition = AddModelDeclarationIfDoesntExit(currentNode, propertyDefinitionSchema, className, targetNamespace, null, true);
+                                        definition = AddModelDeclarationIfDoesntExist(currentNode, propertyDefinitionSchema, className, targetNamespace, null, !propertyDefinitionSchema.IsReferencedSchema());
```

Comment:
Only set to true if we don't have schema. The `targetNamespace` should ideally contain the type if the schema is referenced as the namespace is consistently evaluated from the schema (if it exists). 
Leaving to true results to flaky generation of Types as sometimes it might resolve something like `Microsoft.Graph.Group` vs `Microsoft.Graph.TermStore.Group` depending on which namespace is dug up first(if set to true).